ext2/ext2.c

```diff
@@ -273,13 +300,12 @@ static ssize_t _write(
     {
         for (i = blkno, rem = size, ptr = (uint8_t*)data; rem; i++)
         {
-            uint8_t blk[MYST_BLKSIZE];
             uint32_t off; /* offset into this block */
             uint32_t len; /* bytes to write from this block */
 
             /* Fetch the block */
-            if (dev->get(dev, i, blk) != 0)
-                return -1;
+            if (dev->get(dev, i, v->blk) != 0)
```

Comment:
We should only need to malloc `v->blk` for the last block in this branch.

Reply:
I was aware of this case but wanted to address in phase-2 due to risk. But I did update the PR in this case because it is relatively easy to verify (fingers crossed). :)